### PR TITLE
Fix wrong loggedIn state with 'user: false'

### PR DESCRIPTION
**Pull request by t-matsu-j**
Hi all.

This PR fixes `loggedIn` wrong state with local scheme config `user: false`.
This PR is another solution of #240 .

#240 seems stuck.
Please merge this ASAP.

thank you.

### CODE REVIEW

**Comment by nrm176** [comment]
@pi0 FYI

File: lib/core/auth.js [hunk 1]
```diff
@@ -246,7 +246,9 @@ export default class Auth {
   }
 
   setUser (user) {
-    this.$storage.setState('loggedIn', Boolean(user))
+    const loggedIn = typeof user === 'object' ? Boolean(Object.keys(user).length) : Boolean(user)
```
Comment (manniL) [inline lib/core/auth.js]:
```suggestion
    const loggedIn = Boolean(typeof user === 'object' ? Object.keys(user).length : user)
```

**Comment by t-matsu-j** [comment]
manniL thank you!

**Comment by t-matsu-j** [comment]
Sorry, this works wrong.

I'm fixing up. Give me a time.

**Comment by glennjacobs** [comment]
Please see my comment [here](https://github.com/nuxt-community/auth-module/issues/213#issuecomment-490654746) which I believe is the correct fix for the inconsistent loggedIn state.

**Comment by t-matsu-j** [comment]
@glennjacobs LGTM thanks

now, what should I do?
I'm beginner. Thank you for your helping.

**Comment by t-matsu-j** [comment]
Ah, I found #346. Thank you.

File: lib/core/auth.js [hunk 2]
```diff
@@ -246,7 +246,9 @@ export default class Auth {
   }
 
   setUser (user) {
-    this.$storage.setState('loggedIn', Boolean(user))
+    const loggedIn = Boolean(typeof user === 'object' ? Object.keys(user).length : user)
```
Comment (rchl) [inline lib/core/auth.js]: `null` also has `object` type so you should check for that.

**Comment by pi0** [comment]
@t-matsu-j I really appreciate your help on this PR. Closing it in the flavor of #346. It will be merged and released ASAP.

**Comment by pi0** [comment]
This should be fixed in [v4.6.0](https://github.com/nuxt-community/auth-module/releases/tag/v4.6.0).

**Pull request closed without merging**